data/multiset.lean

```diff
@@ -2884,4 +2884,32 @@ lemma naturality {G H : Type* → Type*}
 quotient.induction_on x
 (by intro; simp [traverse,is_lawful_traversable.naturality] with functor_norm)
 
+section choose
+variables (p : α → Prop) [decidable_pred p] (l : multiset α)
+
+def choose_x : Π hp : (∃! a, a ∈ l ∧ p a), { a // a ∈ l ∧ p a } :=
+quotient.rec_on l (λ l' ex_unique, list.choose_x p l' (exists_of_exists_unique ex_unique)) (begin
+  intros,
+  apply funext,
+  intro hp,
+  suffices all_equal : ∀ x y : { t // t ∈ b ∧ p t }, x = y,
```

Comment:
I guess you want to use `subtype.eq` (or `subtype.eq`) here?!

Reply:
I don't see how?